### ISSUE TITLE
BUG: Series constructor loses date unit of Timestamps

### ISSUE THREAD

**Issue by WillAyd** [pandas-dev/pandas#58300]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
>>> pd.Timestamp("2024-01-01 00:00:00").unit
's'
>>> pd.Series([pd.Timestamp("2024-01-01 00:00:00")]).dt.unit
'ns'
```
```


### Issue Description

I would expect the Series to maintain the unit of 's'

### Expected Behavior

I would expect the Series to maintain the unit of 's'

### Installed Versions

'3.0.0.dev0+681.g434fda08cf'

**Comment by WillAyd** [comment]
@jbrockmendel in case you have any insights into this

**Comment by jbrockmendel** [comment]
Addressed by #55901
